miner/algo_common.go

```diff
@@ -240,15 +247,23 @@ func (envDiff *environmentDiff) commitTx(tx *types.Transaction, chData chainData
 
 // Commit Bundle to env diff
 func (envDiff *environmentDiff) commitBundle(bundle *types.SimulatedBundle, chData chainData, interrupt *int32, algoConf algorithmConfig) error {
-	coinbase := envDiff.baseEnvironment.coinbase
-	tmpEnvDiff := envDiff.copy()
+	var (
+		coinbase   = envDiff.baseEnvironment.coinbase
+		tmpEnvDiff = envDiff.copy()
+
+		coinbaseBalanceBefore = tmpEnvDiff.state.GetBalance(coinbase)
+
+		profitBefore = new(big.Int).Set(tmpEnvDiff.newProfit)
 
-	coinbaseBalanceBefore := tmpEnvDiff.state.GetBalance(coinbase)
+		gasUsed uint64
 
-	profitBefore := new(big.Int).Set(tmpEnvDiff.newProfit)
-	var gasUsed uint64
+		hasCommitErr     bool
+		hasReceiptFailed bool
```

Comment:
Is `hasReceiptFailed` used in this scope? If not, it should not be defined here and instead in the scope it's actually necessary in. Same for `hasCommitErr` and `isRevertibleTx`

Reply:
it's interesting to define them in the scope of a for loop, but the compiler will take care of it so i can update

in this case i'll do it but there are cases where what you're describing would be less than ideal

Reply:
updated